src/Razor/src/rzls/Program.cs

```diff
@@ -76,12 +76,17 @@ public static async Task Main(string[] args)
         ITelemetryReporter? devKitTelemetryReporter = null;
         if (!telemetryExtensionPath.IsNullOrEmpty())
         {
-            using var exportProvider = await ExportProviderBuilder.CreateExportProviderAsync(
-              telemetryExtensionPath).ConfigureAwait(true);
+            using var exportProvider = await ExportProviderBuilder
+                .CreateExportProviderAsync(telemetryExtensionPath)
+                .ConfigureAwait(true);
 
             // Initialize the telemetry reporter if available
             devKitTelemetryReporter = exportProvider.GetExports<ITelemetryReporter>().SingleOrDefault()?.Value;
-            devKitTelemetryReporter?.InitializeSession(telemetryLevel, sessionId, isDefaultSession: true);
+
+            if (devKitTelemetryReporter is ITelemetryReporterInitializer initializer)
```

Comment:
I believe we'll need to make sure we update the telemetry version when we insert into vscode-csharp. @allisonchou is that right?

Reply:
Why would we need to update the telemetry version? This interface is defined in `MS.ANC.Razor.ProjectEngineHost`.

Reply:
Telemetry is technically a separate package since it's devkit only. So I believe we'll just have to make sure we insert the two correctly. It may already be standard practice in VS Code insertions, I was just making a note so we don't forget. Allison or @davidwengier would know more (and hopefully I will soon)

Reply:
By telemetry version, I don't mean MS.VS.Telemetry, I mean our inserted devkit package to vs code

Reply:
> By telemetry version, I don't mean MS.VS.Telemetry, I mean our inserted devkit package to vs code

Ah, yes. They'll need to go together. I hope we're always inserting these together. Not doing that automatically would result in lots of future annoyances for ourselves. :smile:

Reply:
Last time I did an insertion I asked about this, and we don't currently bump our telemetry DLL when we bump Razor. Perhaps that was the wrong answer, but if so, I'm not sure why we have a separate DLL. Maybe at one time rzls was a package that we produced and had a standalone purpose, but now I think its really just Microsoft.Razor.VSCode and there isn't any point pretending otherwise.

Reply:
We still have to keep the devkit package separate because we can't publish MS.VS.Telemetry in the vscode-csharp package. It has to be downloaded at runtime and only if devkit is installed.

Reply:
Yep, I forgot it was a runtime download. The package is self-contained, so it includes the right versions of its dependencies, but if its loaded in the rzls process, then I would imagine we have to bump both at the same time, every time. We can probably tweak the gulp script to that easier in future.

Reply:
Are we all talking about the same thing? I'm confused by the discussion of "our telemetry dll". 😄

Reply:
From my understanding "our telemetry dll" = [Microsoft.VisualStudio.DevKit.Razor.csproj](https://github.com/dotnet/razor/blob/main/src/Razor/src/Microsoft.VisualStudio.DevKit.Razor/Microsoft.VisualStudio.DevKit.Razor.csproj).

Last time we bumped just our server version without bumping our telemetry version in VS Code. It didn't seem to break anything, but since the telemetry dll is being loaded by the rzls process, it might be safer to just bump both like David said.

Reply:
Basically think of it as an external DLL we download when devkit is installed, and has to be MEF imported. The fact that both the contract AND implementation live here is ease of engineering. 

Agree it's best to just update our script to automate updating both always. The worst case would be that when a user updates the extension they have to re-download the same bits they already have because we bumped version.

Reply:
Thanks for the clarification. It was a bit of tribal knowledge that "the telemetry dll" is actually "DevKit.Razor". Any reason we didn't choose a name like `Microsoft.VisualStudio.DevKit.Razor.Telemetry"? Or, do we expect this to contain more things?

Reply:
It is an "it could" and "it only ships with devkit", so naming matched intent.

Reply:
Yeah, it's an "it could" - for example Roslyn has a [similar project](https://github.com/dotnet/roslyn/tree/main/src/VisualStudio/DevKit/Impl), but it contains both their EnC and telemetry logic for DevKit

Reply:
Got it. It probably goes without saying that we should probably match the lingo that we refer to that binary with the intent. So, this isn't "the telemetry dll"; it's our devkit-specific dll. 😉